Add steps for running sysbench against YSQL. (#4300)
yugabyte/yugabyte-db: docs/content/latest/benchmark/sysbench-ysql.md

---
title: Benchmark YSQL performance using Sysbench
headerTitle: Sysbench
linkTitle: Sysbench
description: Benchmark YugabyteDB YSQL performance using Sysbench.
headcontent: Benchmark YugabyteDB YSQL performance using Sysbench.
menu:
  latest:
    identifier: sysbench-ysql
    parent: benchmark
    weight: 5
aliases:
  - /benchmark/sysbench/
showAsideToc: true
isTocNested: true
---
<ul class="nav nav-tabs-alt nav-tabs-yb">

  <li >

>     <a href="/latest/benchmark/tpcc-ysql/" class="nav-link active">

Reviewer: ```suggestion
    <a href="/latest/benchmark/sysbench-ysql/" class="nav-link active">
```

Author: Done.

      <i class="icon-postgres" aria-hidden="true"></i>
      YSQL
    </a>
  </li>

</ul>

> Sysbench is a popular tool for benchmarking system capabilities like CPU, memory and I/O as well as databases like Postgres and MySQL.

Reviewer: ```suggestion
Sysbench is a popular tool for benchmarking databases like postgres and MySQL, as well as system capabilities like CPU, memory and I/O.
```

Author: Done

Follow the steps below to run sysbench against YugabyteDB.

> The version we are using is forked from the [official](https://github.com/akopytov/sysbench) version with a few additions to better reflect YugabyteDB's distributed nature.

Reviewer: ```suggestion
The version we are using is forked from the [official](https://github.com/akopytov/sysbench) version with a few modifications to better reflect YugabyteDB's distributed nature.
```

Author: Done.

> ## Step 1. Get Sysbench.

Reviewer: ```suggestion
## Step 1. Install Sysbench.
```

Author: Done

You can do this by running the following commands.

```sh
$ cd $HOME
$ git clone https://github.com/yugabyte/sysbench.git
$ cd sysbench

> $ ./autogen.sh && ./configure --with-pgsql && make -j && sudo make install

Reviewer: Not needed for this PR but we should make `--with-pgsql` the default in our fork.

Author: Sure.

```

{{< note title="Note" >}}

> The above steps installs the sysbench utility in '/usr/local/bin'

Reviewer: ```suggestion
The above steps will install the sysbench utility in '/usr/local/bin'
```

Author: Done

{{< /note >}}

## Step 2. Start YugabyteDB

Start your YugabyteDB cluster by following the steps in [Quick start](https://docs.yugabyte.com/latest/quick-start/explore-ysql/).

## Step 3. Run the benchmark

> 

Reviewer: Maybe add a header here: `Run all OLTP workloads`

Author: Done.

There is a handly shell script 'run_sysbench.sh' that loads the data and runs the various workloads.
```sh

> ./run_sysbench.sh

Reviewer: Please add a note on how to specify IP address for this script.

Author: Done.

```

> ## Manually run the workloads

Reviewer: ```suggestion
### Manually run the workloads
```

Author: Done.

Before starting the workload we need to load the data first.

```sh

> sysbench oltp_read_only --table-size=1000000 --range_key_partitioning=true --serial_cache_size=1000 --db-driver=pgsql --pgsql-host=127.0.0.1 --pgsql-port=5433 --pgsql-user=yugabyte --pgsql-db=yugabyte prepare

Reviewer: ```suggestion
sysbench oltp_point_select --table-size=1000000 --range_key_partitioning=true --serial_cache_size=1000 --db-driver=pgsql --pgsql-host=127.0.0.1 --pgsql-port=5433 --pgsql-user=yugabyte --pgsql-db=yugabyte prepare
```

Author: Done.

```

Then we can run the workload as follows

```sh

> sysbench oltp_read_only --table-size=1000000 --range_key_partitioning=true --db-driver=pgsql --pgsql-host=127.0.0.1 --pgsql-port=5433 --pgsql-user=yugabyte --pgsql-db=yugabyte --threads=64 --time=120 --warmup-time=120 run

Reviewer: ```suggestion
sysbench oltp_point_select --table-size=1000000 --range_key_partitioning=true --db-driver=pgsql --pgsql-host=127.0.0.1 --pgsql-port=5433 --pgsql-user=yugabyte --pgsql-db=yugabyte --threads=64 --time=120 --warmup-time=120 run
```

Author: Done.
